### ISSUE TITLE
Train cehrgpt fine-tuned R1

### ISSUE THREAD

**Issue by ChaoPang** [knatarajan-lab/cehrgpt#47]
We use reinforcement learning to finetune cehrgpt base V2 using a set of reward functions derived from the CUIMC source data statistics to create cehrgpt fine-tuned R1. 

- The marginal distribution of the concepts
- The conditional probability distribution
- The average length of the sequences in the generated batch
- The generated sequences need to be a valid sequence
